ios/RNSScreen.mm

```diff
@@ -602,58 +809,297 @@ - (BOOL)isTransparentModal
+        sheetLudLimitExclusive = 2;
+
+        if (first < second) {
+          [self setAllowedDetentsForSheet:sheet
+                                       to:@[
+                                         UISheetPresentationControllerDetent.mediumDetent,
+                                         UISheetPresentationControllerDetent.largeDetent
+                                       ]
+                                  animate:YES];
+        } else {
+          RCTLogError(@"The values in sheetAllowedDetents array must be sorted");
+        }
       } else {
-        RCTLogError(@"Unhandled value of sheetLargestUndimmedDetent passed");
+        float first = _sheetAllowedDetents[0].floatValue;
+        if (first == SHEET_FIT_TO_CONTENTS) {
+          RCTLogError(@"Unsupported on iOS versions below 16");
+        } else if (first < 1.0) {
+          [self setAllowedDetentsForSheet:sheet to:@[ UISheetPresentationControllerDetent.mediumDetent ] animate:YES];
+          [self setSelectedDetentForSheet:sheet to:UISheetPresentationControllerDetentIdentifierMedium animate:YES];
+        } else {
+          [self setAllowedDetentsForSheet:sheet to:@[ UISheetPresentationControllerDetent.largeDetent ] animate:YES];
+          [self setSelectedDetentForSheet:sheet to:UISheetPresentationControllerDetentIdentifierLarge animate:YES];
+        }
       }
+    }
 
-      if (_sheetAllowedDetents == RNSScreenDetentTypeMedium) {
-        sheet.detents = @[ UISheetPresentationControllerDetent.mediumDetent ];
-        if (sheet.selectedDetentIdentifier != UISheetPresentationControllerDetentIdentifierMedium) {
-          [sheet animateChanges:^{
-            sheet.selectedDetentIdentifier = UISheetPresentationControllerDetentIdentifierMedium;
-          }];
+    sheet.prefersScrollingExpandsWhenScrolledToEdge = _sheetExpandsWhenScrolledToEdge;
+    [self setGrabberVisibleForSheet:sheet to:_sheetGrabberVisible animate:YES];
+    [self setCornerRadiusForSheet:sheet to:_sheetCornerRadius animate:YES];
+
+    int detentIndex = _sheetLargestUndimmedDetent != nil ? _sheetLargestUndimmedDetent.intValue : -1;
+    detentIndex = detentIndex >= sheetLudLimitExclusive ? detentIndex - 1 : detentIndex;
+    if (detentIndex == -1) {
+      [self setLargestUndimmedDetentForSheet:sheet to:nil animate:YES];
+    } else if (detentIndex >= 0) {
+      if (systemDetentsInUse) {
+        if (sheetLudLimitExclusive == 0 || (sheetLudLimitExclusive == 1 && _sheetAllowedDetents[0].floatValue < 1.0)) {
+          [self setLargestUndimmedDetentForSheet:sheet
+                                              to:UISheetPresentationControllerDetentIdentifierMedium
+                                         animate:YES];
+        } else {
+          [self setLargestUndimmedDetentForSheet:sheet
+                                              to:UISheetPresentationControllerDetentIdentifierLarge
+                                         animate:YES];
```

Comment:
This is quite unreadable without the context. Let's add some comments what is going on here (maybe let's separate if conditions from the most inner if?

Also, why are we redeclaring `detentIndex`? Maybe let's separate logic to two independent variables?

Reply:
> Also, why are we redeclaring detentIndex? Maybe let's separate logic to two independent variables?

Don't really get your question here, but I've renamed the `detentIndex` variable to `largestUndimmedDetentIndex` (in short `ludIndex`) so that it's purpose is more clear.

> This is quite unreadable without the context. Let's add some comments what is going on here (maybe let's separate if conditions from the most inner if?

🫡

Reply:
Done, let me know whether it is ok now